applications/IgaApplication/custom_utilities/create_breps_sbm_utilities.h

```diff
@@ -0,0 +1,443 @@
+#include "geometries/point_on_geometry.h"
+
+#include "geometries/nurbs_curve_geometry.h"
+#include "geometries/nurbs_surface_geometry.h"
+
+#include "geometries/brep_surface.h"
+#include "geometries/brep_curve.h"
+
+#include "geometries/nurbs_curve_on_surface_geometry.h"
+#include "geometries/brep_curve_on_surface.h"
+
+
+namespace Kratos
+{
+
+///@name Kratos Classes
+///@{
+template<class TNodeType = Node, class TEmbeddedNodeType = Point>
+class CreateBrepsSBMUtilities : public IO
+{
+    public:
+
+    ///@}
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of CreateBrepsSBMUtilities
+    KRATOS_CLASS_POINTER_DEFINITION(CreateBrepsSBMUtilities);
+
+    using SizeType = std::size_t;
+    using IndexType = std::size_t;
+
+    using GeometryType = Geometry<TNodeType>;
+    using GeometryPointerType = typename GeometryType::Pointer;
+
+    using ContainerNodeType = PointerVector<TNodeType>;
+    using ContainerEmbeddedNodeType = PointerVector<TEmbeddedNodeType>;
+
+    using CouplingGeometryType = CouplingGeometry<TNodeType>;
+
+    using NurbsSurfaceType = NurbsSurfaceGeometry<3, ContainerNodeType>;
+    using NurbsTrimmingCurveType = NurbsCurveGeometry<2, ContainerEmbeddedNodeType>;
+
+    using NurbsSurfacePointerType = typename NurbsSurfaceType::Pointer;
+    using NurbsTrimmingCurvePointerType = typename NurbsTrimmingCurveType::Pointer;
+
+    using NurbsSurfaceGeometryType = NurbsSurfaceGeometry<3, PointerVector<NodeType>>;
+    using NurbsSurfaceGeometryPointerType = typename NurbsSurfaceGeometryType::Pointer;
+
+    using BrepSurfaceType = BrepSurface<ContainerNodeType, true, ContainerEmbeddedNodeType>;
+    using BrepCurveOnSurfaceType = BrepCurveOnSurface<ContainerNodeType, true, ContainerEmbeddedNodeType>;
+
+    using BrepCurveType = BrepCurve<ContainerNodeType, ContainerEmbeddedNodeType>;
+    using PointOnGeometryOnSurfaceType = PointOnGeometry<ContainerNodeType, 3, 2>;
+    using PointOnGeometryOnCurveType = PointOnGeometry<ContainerNodeType, 3, 1>;
+
+    using BrepCurveOnSurfaceArrayType = DenseVector<typename BrepCurveOnSurfaceType::Pointer>;
+    using BrepCurveOnSurfaceLoopType = DenseVector<typename BrepCurveOnSurfaceType::Pointer>;
+    using BrepCurveOnSurfaceLoopArrayType = DenseVector<DenseVector<typename BrepCurveOnSurfaceType::Pointer>>;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Constructor with path to input file.
+    CreateBrepsSBMUtilities(
+        SizeType EchoLevel = 0)
+        : mEchoLevel(EchoLevel)
+    {
+    }
+
+    /// Destructor.
+    ~CreateBrepsSBMUtilities() = default;
+
+    ///@}
+    ///@name Python exposed Functions
+    ///@{
+
+    /// Adds the surface geometry to the herin provided model_part and create the boundary breps for the SBM case.
+    void CreateSurrogateBoundary(NurbsSurfaceGeometryPointerType& p_surface, ModelPart& rModelPart, ModelPart& rSurrogateModelPartInner, ModelPart& rSurrogateModelPartOuter, const Point& A_uvw, const Point& B_uvw)

```

Comment:
```suggestion
    void CreateSurrogateBoundary(
        NurbsSurfaceGeometryPointerType& rSurface,
        ModelPart& rModelPart,
        ModelPart& rSurrogateModelPartInner,
        ModelPart& rSurrogateModelPartOuter,
        const Point& rPointAuvw,
        const Point& rPointBuvw)
```
Please follow the style guide. Do so for the rest of the PR. Once you finish I'll review the technicalities.